src/main/java/hudson/security/ProjectMatrixAuthorizationStrategy.java

```diff
@@ -70,7 +71,7 @@ public ACL getACL(Job<?,?> project) {
 
     private static ACL inheritingACL(final ACL parent, final ACL child) {
         if (parent instanceof SidACL && child instanceof SidACL) {
-            return ((SidACL) parent).newInheritingACL((SidACL) child);
+            return ((SidACL) child).newInheritingACL((SidACL) parent);
```

Comment:
Amends a change made in #8. @stephenc I believe you got the receiver and argument backward when you did that.